subworkflows/local/channel_chunks_create_csv/main.nf

```diff
@@ -0,0 +1,17 @@
+workflow CHANNEL_CHUNKS_CREATE_CSV {
```

Comment:
I don't know if all those CSV creation needs to be separate subworkflows, they could be separate functions.
Putting them into a function would also allow to factorize a lot of the process and simplify greatly how it's working.

Reply:
Do you have a similar example in a pipeline?

Reply:
Not really.
But this is something I've been thinking for a while and this is precisely the case that could take advantage of that.

Reply:
I understand your suggestion and we have to keep this in mind for future improvements. However, given the differences in the elements of the metamap and the varying locations where the final files are saved depending on the tool, it might be challenging to implement a single function that accommodates all these variations effectively.

This current approach ensures that each subworkflow handles its specific requirements accurately. This could be an initial step, and we can iteratively refine this in further versions when there will be no major changes in the code to possibly factor out common patterns as we better understand the variations.

Reply:
I agree, we need to add that to the roadmap 😉